pkg/format/format.go

```diff
@@ -143,24 +165,31 @@ func findDrives() {
 		// this probably shouldn't be so hard
 		// but d.Mode()&os.ModeDevice == 0 doesn't work as expected
 		mode := d.Sys().(*syscall.Stat_t).Mode
+
 		if (mode & syscall.S_IFMT) != syscall.S_IFBLK {
 			continue
 		}
+
 		// ignore if it matches regexp
 		if ignoreExp.MatchString(d.Name()) {
 			continue
 		}
-		driveKeys = append(driveKeys, filepath.Join("/dev", d.Name()))
-	}
-	sort.Strings(driveKeys)
-	for _, d := range driveKeys {
-		drives[d] = true
+
+		driveKey := filepath.Join("/dev", d.Name())
```

Comment:
factored the `driveKey` out of the prior `append` so I can compare it with the value of the `--device` argument (if applicable)